### ISSUE TITLE
Cookie used to access presence api times out after an hour

### ISSUE THREAD

**Issue by chrisfinch** [guardian/about-to-be-published#4]
What could we do about this?

cc @lindseydew

**Comment by lindseydew** [comment]
Yeah, this is a bit tricky. 

Workflow handles this by implementing some logic which will reauth under the hood, but don't think this will be trivial for us to implement, especially as we are a 'read-only' auth service.

cc @jamespamplin, @jamesgorrie @theefer - do you guys have any clever client side tricks for handling authentication expiry?

**Comment by jamesgorrie** [comment]
If you're using pan-domain you can use [`panda-session`](https://github.com/guardian/panda-session)

**Comment by jamesgorrie** [comment]
There is also [pandular](https://github.com/guardian/pandular) if you're using angular.

You will need a login URL, which might be able to be set to the Workflow URL that you're using in the first place.

It\'s be nice to have a `panda.gutools.co.uk` service to be able to login from all client-side only apps.

**Comment by piuccio** [comment]
Or `login.gutools.co.uk`.

Fronts tool is using `panda-session`, there are some edge cases to consider, but it mostly works fine.

**Comment by jamesgorrie** [comment]
> `login.gutools.co.uk`

Doesn't have the same ring to it.

**Comment by theefer** [comment]
Yes you should be able to use the panda-session library and call it whenever you get a 419, although I can't remember whether it requires the auth to redirect to a page on the same domain as you so it can know the reauth was successful. Happy to have a chat about this as I've been working on related stuff in the Grid recently, and we want to do the same for our grid-inspector.

**Comment by lindseydew** [comment]
Cool, if I'll come have a chat with you if I get chance to pick this up.
